internal/license/service.go

```diff
@@ -129,11 +132,54 @@ func InjectCustomLicenseBytes(res *service.Resources, conf Config, licenseBytes
 		"expires_at", time.Unix(license.Expiry, 0).Format(time.RFC3339),
 	).Info("Successfully loaded Redpanda license")
 
-	s.loadedLicense.Store(&license)
+	s.setLicense(res, &license)
 	setSharedService(res, s)
+
 	return nil
 }
 
+func (s *Service) setLicense(res *service.Resources, l *RedpandaLicense) {
+	s.loadedLicense.Store(l)
+
+	if s.cancel != nil {
+		s.cancel()
+	}
+	if l == nil || !l.AllowsEnterpriseFeatures() {
+		return
+	}
+
+	if s.expiryMetric == nil {
+		s.expiryMetric = res.Metrics().NewGauge("redpanda_cluster_features_enterprise_license_expiry_sec",
+			"Seconds remaining until the enterprise license expires.",
+		)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
```

Comment:
I guess this doesn't get cleaned up, but I guess that's fine for now (and would take a bunch of wiring to cleanup for unit tests)?

Alternatively, there is a seperate method to start the monitoring loop that we can run in the CLI main method and then tests aren't effected.